src/Symfony/Component/Workflow/Dumper/GraphvizDumper.php

```diff
@@ -84,7 +84,7 @@ private function findTransitions(Definition $definition)
         $transitions = array();
 
         foreach ($definition->getTransitions() as $transition) {
-            $transitions[] = array(
+            $transitions[$transition->getName()] = array(
```

Comment:
If they have the same name, they have different from/to, so they should both be rendered on the graph separately (otherwise, the graph will lie).

Btw, for a finite state machine, we could change the graph to render transitions as edges directly (optionally, as rendering the graph in workflow mode is still useful), as they are known to have a single from and a single to